main.go

```diff
@@ -78,19 +74,18 @@ func main() {
 		linktype = layers.LinkTypeRaw
 	}
 	if err = errors.WithStack(pcapw.WriteFileHeader(1600, linktype)); err != nil {
-		return
+		exitWithErr(errors.WithStack(err))
 	}
 
 	skbw, e := os.Create(config.SkbFilename)
 	if e != nil {
-		err = errors.WithStack(e)
-		return
+		exitWithErr(errors.WithStack(e))
 	}
 	defer skbw.Close()
 
 	host, err := sysinfo.Host()
 	if err != nil {
-		return
+		exitWithErr(errors.WithStack(err))
```

Comment:
For instance, if the process calls `log.fatal` here, the `defer delIngress()` and `defer delEgress()` won't execute as expect. This is crucial as we don't want to leave any tc filter on devices after tracing.

Reply:
You are right. I forgot these defer functions.